include/boost/multiprecision/detail/standalone_config.hpp

```diff
@@ -0,0 +1,71 @@
+///////////////////////////////////////////////////////////////
+//  Copyright 2010 - 2021 Douglas Gregor
+//  Copyright 2021 Matt Borland.
+//  Distributed under the Boost Software License, Version 1.0.
+//  See accompanying file LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
+//
+//  Used to support configuration options depending on standalone context
+//  by providing either required support or disabling functionality  
+
+#ifndef BOOST_MP_STANDALONE_CONFIG_HPP
+#define BOOST_MP_STANDALONE_CONFIG_HPP
+
+// If any of the most frequently used boost headers are missing assume that standalone mode is supposed to be used
+#ifdef __has_include
+#if !__has_include(<boost/config.hpp>) || !__has_include(<boost/assert.hpp>) || !__has_include(<boost/lexical_cast.hpp>) || \
+    !__has_include(<boost/throw_exception.hpp>) || !__has_include(<boost/predef/other/endian.h>)
+#   ifndef BOOST_MP_STANDALONE
+#       define BOOST_MP_STANDALONE
+#   endif
+#endif
+#endif
+
+#ifndef BOOST_MP_STANDALONE
+
+#include <boost/config.hpp>
+#include <boost/integer.hpp>
+#include <boost/integer_traits.hpp>
+
+#else // Standalone mode
+
+// Prevent Macro sub
+#ifndef BOOST_PREVENT_MACRO_SUBSTITUTION
+#  define BOOST_PREVENT_MACRO_SUBSTITUTION
+#endif
+
+// int/float 128 types
+
+#ifdef (__SIZEOF_INT128__)
+#  define BOOST_HAS_INT128
+#endif
+
+#ifdef (__SIZEOF_FLOAT128__)
+#  define BOOST_HAS_FLOAT128
+#endif
+
+#if defined(BOOST_HAS_INT128) && defined(__cplusplus)
+namespace boost{
```

Comment:
I think we should avoid putting these directly in namespace boost: imagine 2 different Boost libraries doing the same thing.  Suggest we put these directly into boost::multiprecision here, and remove the definitions in cpp_int_config.hpp (or vice versa?)